extensions/blocks/simple-payments/utils.js

```diff
@@ -21,9 +22,50 @@ export const decimalPlaces = number => {
 	return Math.max( 0, ( match[ 1 ] ? match[ 1 ].length : 0 ) - ( match[ 2 ] ? +match[ 2 ] : 0 ) );
 };
 
-export const formatPrice = ( price, currency, withSymbol = true ) => {
+const getNavigatorLanguage = () => {
+	if ( navigator.languages && navigator.languages.length ) {
+		return navigator.languages[ 0 ];
+	}
+	return navigator.userLanguage || navigator.language || navigator.browserLanguage || 'en';
+};
+
+// Legacy method of displaying prices.
+export const formatPriceFallback = ( price, currency, withSymbol = true ) => {
 	const { precision, symbol } = getCurrencyDefaults( currency );
 	const value = price.toFixed( precision );
 	// Trim the dot at the end of symbol, e.g., 'kr.' becomes 'kr'
 	return withSymbol ? `${ value } ${ trimEnd( symbol, '.' ) }` : value;
 };
+
+// Display prices using Intl.NumberFormat - supported in 95.75% of browsers as of Oct 2020.
+const formatPriceIntlNumberFormat = ( price, currency, withSymbol = true ) => {
+	const { siteLocale } = select( 'core/block-editor' ).getSettings();
```

Comment:
`const { siteLocale } = select( 'core/block-editor' ).getSettings();` will return the current user interface language and not the current site language. So, when a user (me) has `en` as interface language (very common) but `greek` as site language then the symbol appears on the left whereas it should appear in the right.

![](https://cln.sh/ks7xdO+)

I searched it a bit, I am not sure if `getLocaleSlug` from `i18n-calypso` will return the correct locale
https://github.com/Automattic/wp-calypso/blob/aa011d0e42b1190ead34a407538d21e7e8fe2186/packages/i18n-calypso/src/i18n.js#L314

Reply:
Good catch! I pushed an update that uses getLocalSlug, can you try it out and let me know if it works? I tried updating both my site and wordpress user language to Greek, and it didn't seem to work.  Maybe it's because I'm using the wp-admin edit to test?  This may be difficult..

Reply:
It still returns the `en` version for me too. Seems like 
https://github.com/Automattic/wp-calypso/blob/aa011d0e42b1190ead34a407538d21e7e8fe2186/packages/i18n-calypso/src/i18n.js#L256-L259
it returns the `defaultLocaleSlug`.... That said, I am not sure how we can fix this.